processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -144,15 +161,13 @@ public SearchQueryQueryToolChest(
         final byte[] querySpecBytes = query.getQuery().getCacheKey();
         final byte[] granularityBytes = query.getGranularity().cacheKey();
 
-        final Collection<DimensionSpec> dimensions = query.getDimensions() == null
-                                                     ? ImmutableList.<DimensionSpec>of()
-                                                     : query.getDimensions();
-
-        final byte[][] dimensionsBytes = new byte[dimensions.size()][];
+        final List<DimensionSpec> dimensionSpecs =
```

Comment:
Seems like it could if it wants, since for all call sites, the two `query` objects are the same.